accounts/migrations/0001_initial.py

```diff
@@ -20,7 +20,7 @@ class Migration(migrations.Migration):
                 ('password', models.CharField(max_length=128, verbose_name='password')),
                 ('last_login', models.DateTimeField(default=django.utils.timezone.now, verbose_name='last login')),
                 ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
-                ('username', models.CharField(error_messages={'unique': 'A user with that username already exists.'}, max_length=255, validators=[django.core.validators.RegexValidator('^[\\w.@+-]+$', 'Enter a valid username. This value may contain only letters, numbers and @/./+/-/_ characters.')], help_text='Required. 255 characters or fewer. Letters, digits and @/./+/-/_ only.', unique=True, verbose_name='username')),
+                ('username', models.CharField(error_messages={'unique': 'A user with that username already exists.'}, max_length=255, validators=[django.contrib.auth.validators.ASCIIUsernameValidator()], help_text='Required. 255 characters or fewer. Letters, digits and @/./+/-/_ only.', unique=True, verbose_name='username')),
```

Comment:
Ditto. An example would help understand how a spurious migration is mitigated.

Reply:
Hi @zmousm 

When I revert this commit and run `./manage.py makemigrations`, I get:
```
Migrations for 'accounts':
  accounts/migrations/0005_auto_20190225_1059.py
    - Alter field username on user
```
with `accounts/migrations/0005_auto_20190225_1059.py` having:
```
# -*- coding: utf-8 -*-
# Generated by Django 1.11.20 on 2019-02-24 21:59
from __future__ import unicode_literals

import django.contrib.auth.validators
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('accounts', '0004_alter_user_last_login_null'),
    ]

    operations = [
        migrations.AlterField(
            model_name='user',
            name='username',
            field=models.CharField(error_messages={'unique': 'A user with that username already exists.'}, help_text='Required. 255 characters or fewer. Letters, digits and @/./+/-/_ only.', max_length=255, unique=True, validators=[django.contrib.auth.validators.ASCIIUsernameValidator()], verbose_name='username'),
        ),
    ]
```
which does not really change anything (no changes to the DB), but is recorded as a new migration, re-iterating all of the user model properties.  And it got triggered by the properties of UserModel in our copy not exactly matching what upstream evolved to.  So that's why I changed our copy of the user model to match upstream.

The other commit prevents pretty much exactly the same thing - both had to be in to make our model exactly match upstream.

Reply:
I see. What is even worse is that the username validator also changes based on whether it is running on Python 3...

While such migrations are noop and this is mostly an annoyance, I'm not sure what the most appropriate approach would be here, especially considering the runtime aspect above.

I need to investigate further...

Reply:
So yes the need to adapt to Django's changes to `django.contrib.auth.models.AbstractUser` is probably necessary; it would seem to be the price we pay for extending a complete class rather than `AbstractBaseUser`.

However I believe we should tweak the migration like https://github.com/django/django/commit/526575c64150e10dd8666d1ed3f86eedd00df2ed so as to avoid more spurious migrations sparked by a change in the run-time environment (PY3, eventually). I will submit a PR for that.